spec/features/schemes_spec.rb

```diff
@@ -623,6 +630,20 @@
               expect(page).to have_current_path("/schemes/#{scheme.id}/check-answers")
               expect(page).to have_content "Check your changes before creating this scheme"
             end
+
+            it "keeps the provider answer when swithing between other provider options" do
+              click_link("Change", href: "/schemes/#{scheme.id}/details?check_answers=true", match: :first)
+              choose "Another organisation"
+              click_button "Save and continue"
+              expect(page).to have_content(another_organisation.name)
+            end
+
+            it "does not display the answer if it's changed to the same support provider" do
+              click_link("Change", href: "/schemes/#{scheme.id}/details?check_answers=true", match: :first)
+              choose "The same organisation that owns the housing stock"
+              click_button "Save and continue"
+              expect(page).not_to have_content("Organisation providing support")
```

Comment:
It would be good to have controller request test, that will investigate the state of the scheme after selecting the same org ( basically making sure that values are updated in DB ).